docs/experimental-features.md

```diff
@@ -0,0 +1,214 @@
+        },
+        {
+          "Protocol": "/ipfs/kad/1.0.0"
+        }
+      ]
+    },
+    {
+...
+```
+
+### Road to being a real feature
+- [ ] Significant real world testing and performance metrics across a wide variety of workloads showing that it works well.
+
+---
+
+## Raw Leaves for unixfs files
+Allows files to be added with no formatting in the leaf nodes of the graph.
+
+### State
+experimental.
+
+### In Version
+master, 0.4.5
+
+### How to enable
+Use `--raw-leaves` flag when calling `ipfs add`.
+
+### Road to being a real feature
+- [ ] Needs more people to use and report on how well it works.
+
+---
+
+## ipfs filestore
+Allows files to be added without duplicating the space they take up on disk.
+
+### State
+experimental.
+
+### In Version
+master, 0.4.7
+
+### How to enable
+Modify your ipfs config:
+```
+ipfs config --json Experimental.FilestoreEnabled true
+```
+
+And then pass the `--nocopy` flag when running `ipfs add`
+
+### Road to being a real feature
+- [ ] Needs more people to use and report on how well it works.
+- [ ] Need to address error states and failure conditions
+- [ ] Need to write docs on usage, advantages, disadvantages
+- [ ] Need to merge utility commands to aid in maintenance and repair of filestore
+
+---
+
+## Private Networks
+
+Allows ipfs to only connect to other peers who have a shared secret key.
+
+### State
+Experimental
+
+### In Version
+master, 0.4.7
+
+### How to enable
+Generate a pre-shared-key using [ipfs-swarm-key-gen](https://github.com/Kubuxu/go-ipfs-swarm-key-gen)):
+```
+go get github.com/Kubuxu/go-ipfs-swarm-key-gen/ipfs-swarm-key-gen
+ipfs-swarm-key-gen > ~/.ipfs/swarm.key
+```
+
+To join a given private network, get the key file from someone in the network and save it to `~/.ipfs/swarm.key` (If you are using a custom `$IPFS_PATH`, put it in there instead).
+
+When using this feature, you will not be able to connect to the default bootstrap nodes (Since we arent part of your private network) so you will need to set up your own bootstrap nodes.
+
+To prevent your node from even trying to connect to the default bootstrap nodes, run:
+```bash
+ipfs bootstrap rm --all
+```
+
+To be extra cautious, You can also set the `LIBP2P_FORCE_PNET` environment variable to `1` to force the usage of private networks. If no private network is configured, the daemon will fail to start.
+
+### Road to being a real feature
+- [ ] Needs more people to use and report on how well it works
+- [ ] More documentation
+
+---
+
+## ipfs p2p
+Allows to tunnel TCP connections through Libp2p sterams
+
+### State
+Experimental
+
+### In Version
+master, 0.4.10
+
+### How to enable
+P2P command needs to be enabled in config
+
+`ipfs config --json Experimental.Libp2pStreamMounting true`
+
+### How to use
+
+Basic usage:
+
+- Open a listener on one node (node A)
+`p2p listener open p2p-test /ip4/127.0.0.1/tcp/10101`
+- Connect your app to the endpoint opened locally (tcp on localhost port 10101)
```

Comment:
This is not exactly correct, I updated the comment in the experimental features issue (https://github.com/ipfs/go-ipfs/issues/3397#issuecomment-309306293)